filestore/test/test_commands.py

```diff
@@ -6,46 +6,39 @@
 import uuid
 import mongoengine
 import mongoengine.connection
-from nose.tools import make_decorator
-from mongoengine.context_managers import switch_db
+
 
 import filestore.commands as fc
 import filestore.retrieve as fsr
-from filestore.odm_templates import Resource, Datum
+from filestore.odm_templates import Datum, ALIAS
 from numpy.testing import assert_array_equal
 from nose.tools import assert_raises
 
 from .t_utils import SynHandlerMod
 
 db_name = str(uuid.uuid4())
-dummy_db_name = str(uuid.uuid4())
+conn = None
 
 
 def setup():
-    # need to make 'default' connection to point to no-where, just to be safe
-    # need this to exist so the context managers work
-    mongoengine.connect(dummy_db_name, 'fs')
+    global conn
+    # make sure nothing is connected
+    fc.db_disconnect()
+    # make sure it _is_ connected
+    conn = mongoengine.connect(db_name, host='localhost',
+                        alias=ALIAS)
 
-    # connect to the db we are actually going to use
-    mongoengine.connect(db_name, alias='test_db')
     # register the dummy handler to use
     fsr.register_handler('syn-mod', SynHandlerMod)
 
 
 def teardown():
-    conn = mongoengine.connection.get_connection('test_db')
-    conn.drop_database(db_name)
-    conn.drop_database(dummy_db_name)
-    del fsr._h_registry['syn-mod']
-
+    fc.db_disconnect()
+    # if we know about a connection, drop the database
```

Comment:
because I am not going to modify what is assigned to `conn` and because teardown will correctly search upwards in scope until it finds the module level `conn`